packages/guides/src/Compiler/NodeTransformers/TocNodeTransformer.php

```diff
@@ -80,16 +80,16 @@ private function buildFromSection(
         }
 
         foreach ($entry->getChildren() as $child) {
-            yield from $this->buildLevel($child, $document, $depth, $node);
+            yield from $this->buildLevel($child, $document, $depth, $node, false);
         }
     }
 
     /** @return Traversable<Entry> */
-    private function buildLevel(MetaEntry $child, DocumentEntry $document, int $depth, TocNode $node): Traversable
+    private function buildLevel(MetaEntry $child, DocumentEntry $document, int $depth, TocNode $node, bool $isDocumentRoot): Traversable
     {
         if ($child instanceof SectionEntry) {
             yield new Entry(
-                $document->getFile(),
+                $document->getFile().($isDocumentRoot ? '' : '#'.$child->getId()),
```

Comment:
I think the anchor should be a new property putting all in the same property makes the url building more complex. As the output format is not known by the compiler. The file url is just a reference to find the correct file and render a link.

Reply:
Agreed, I'm not used yet to think about different output formats :)